keeper/src/main/scala/zio/keeper/consensus/Consensus.scala

```diff
@@ -0,0 +1,172 @@
+package zio.keeper.consensus
+
+import java.util.UUID
+
+import zio._
+import zio.clock.Clock
+import zio.keeper.membership.{ ByteCodec, Membership, MembershipEvent, NodeId, TaggedCodec }
+import zio.keeper.{ Error, Message, membership }
+import zio.stream.{ Stream, ZSink }
+import zio.keeper.protocol.InternalProtocol._
+import upickle.default._
+import zio.duration.Duration
+import zio.logging.Logging
+
+object Consensus {
+
+  trait Service {
+    def getLeader: UIO[NodeId]
```

Comment:
I think this service should 

```
trait Service {
   val leader: IO[NoLeaderError, NodeId]
   val onLeaderChange: Stream[Error, NodeId]
}
```

Reply:
rest of the methods are in ```Membership``` which will be required but consensus I believe

Reply:
OK, so then as I understand we have access to all modules from our protocol stack? Not only to the one on the top?

Reply:
Then I also agree

Reply:
Last thing: `getView` is basically an ordered `nodes` from Membership according to the joining time. Remove it also?

Reply:
ping

Reply:
I was thinking about this recently because I read some paper about leader election and what they did to achieve order they use IP addresses so 1.1.1.1 < 1.1.1.2. Maybe we should do similar thing because time in distributed systems is always a bit sneaky because clock can be skew or nodes join in the exactly same time. @mijicd what do you think?

Reply:
Yes, it would be great if we can avoid ordering anything by (wall clock) timestamps.

Reply:
@pshemass Thanks! My side note: So here (for convergence) when a new leader is elected (because previous left) the new leader will broadcast it's view (ordered nodes) which other will always accept even if they would see different node as a leader, or had some messed up order of joined nodes, so (but in some edge cases this approach also might fail I guess). WDYT?

Reply:
@pshemass sure, so as I understand election will be resolved by choosing smallest value of `address` + `port` value among members?